cbor2/types.py

```diff
@@ -62,36 +64,30 @@ def __repr__(self):
         return 'CBORTag({self.tag}, {self.value!r})'.format(self=self)
 
 
-class CBORSimpleValue(object):
+class CBORSimpleValue(namedtuple('CBORSimpleValue', ['value'])):
     """
     Represents a CBOR "simple value".
 
     :param int value: the value (0-255)
     """
 
-    __slots__ = 'value'
+    __slots__ = ()
+    __hash__ = namedtuple.__hash__
 
-    def __init__(self, value):
+    def __new__(cls, value):
         if value < 0 or value > 255:
             raise TypeError('simple value out of range (0..255)')
-        self.value = value
+        return super(CBORSimpleValue, cls).__new__(cls, value)
 
     def __eq__(self, other):
-        if isinstance(other, CBORSimpleValue):
-            return self.value == other.value
-        elif isinstance(other, int):
+        if isinstance(other, int):
             return self.value == other
-        return NotImplemented
+        return super(CBORSimpleValue, self).__eq__(other)
 
     def __ne__(self, other):
```

Comment:
Is this really required? Strikes me as somewhat odd to have `__eq__()` and `__ne__()` both implemented but not `__lt__ ()`. Have you that these operations require these specific implementations? If you haven't already, I suggest you check out `@total_ordering`. I don't know how much the superclass already provides, but these are the points that I wanted to bring up.

Reply:
I will look into `@total_ordering` thanks

Reply:
Looks like I have to implement each comparison operator since `@total_ordering` operator re-orders the objects. If we want to keep the comparison to integers.